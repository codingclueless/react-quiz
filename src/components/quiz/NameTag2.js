import React, { useState } from 'react';
import PropTypes from 'prop-types';
import styles from './NameTag2.css';

const Shape = ({ text, color, backgroundColor }) => {


  return (
    <div style={{ height: '200px', width: '200px', color:color, backgroundColor:backgroundColor }}>
      <h1>{text}</h1>
    </div>
  );
};

Shape.propTypes = {
  text: PropTypes.string.isRequired,
  color: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired
};


const TextSelector = ({ text, changeText }) => {
  
  return (
    <>
      <input type="text" value={text} onChange={changeText} placeholder='Insert text...'/>
    </>
  );
};

TextSelector.propTypes = {
  text: PropTypes.string.isRequired,
  changeText: PropTypes.func.isRequired,
};

const ColorSelector = ({ color, backgroundColor, changeColor, changeBackgroundColor }) => {

  return (
    <>
      <div>
        <h3>Text Color:</h3>
        <input type="color" value={color} onChange={changeColor}/>
      </div>
      <div>
        <h3>Background Color:</h3>
        <input type="color" value={backgroundColor} onChange={changeBackgroundColor}/>
      </div>
    </>
  );
};

ColorSelector.propTypes = {
  color: PropTypes.string.isRequired,
  backgroundColor: PropTypes.string.isRequired,
  changeColor: PropTypes.func.isRequired,
  changeBackgroundColor: PropTypes.func.isRequired,
};

const NameTag = () => {
  const [text, setText] = useState('');
  const [color, setColor] = useState('');
  const [backgroundColor, setBackgroundColor] = useState('');

  const changeText = ({ target }) => {
    setText(target.value);
  };

  const changeColor = ({ target }) => {
    setColor(target.value);
  };

  const changeBackgroundColor = ({ target }) => {
    setBackgroundColor(target.value);
  };

  return (
    <div className={styles.NameTag}>
      <h1>React Quiz</h1>
      <ColorSelector color={color} backgroundColor={backgroundColor} changeColor={changeColor} changeBackgroundColor={changeBackgroundColor} />
      <TextSelector text={text} changeText={changeText} />
      <Shape text={text} color={color} backgroundColor={backgroundColor}/>
    </div>

  );
};

export default NameTag;
