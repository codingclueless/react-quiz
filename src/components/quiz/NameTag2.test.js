import React from 'react';
import { shallow } from 'enzyme';
import NameTag2 from './NameTag2';

describe('NameTag2 component', () => {
  it('renders App', () => {
    const wrapper = shallow(<NameTag2 />);
    expect(wrapper).toMatchSnapshot();
  });
});
