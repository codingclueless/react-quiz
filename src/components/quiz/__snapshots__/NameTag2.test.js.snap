// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NameTag2 component renders App 1`] = `
<div
  className="NameTag"
>
  <ColorSelector
    backgroundColor=""
    changeBackgroundColor={[Function]}
    changeColor={[Function]}
    color=""
  />
  <TextSelector
    changeText={[Function]}
    text=""
  />
  <Shape
    backgroundColor=""
    color=""
    text=""
  />
</div>
`;
